### DISCUSSION TITLE
What's the right argument should be added to host to build for wasm 32-unknown-unknown?

### DISCUSSION THREAD

**Discussion by asmmo** [jedisct1/libsodium#1145] [Q&A] [answered]
As the question asks. please, tell me the right argument?

**Comment by jedisct1** [comment, marked as answer]
Libsodium requires either the `emscripten` or `wasi` environments, as it's not even possible to generate random numbers in standalone WebAssembly.

To compile to WASI, the `host` argument should be `wasm32-wasi`, same as the compiler's target.

See the existing build script: https://github.com/jedisct1/libsodium/blob/master/dist-build/wasm32-wasi.sh

or the Rust code: https://github.com/jedisct1/libsodium-sys-stable/blob/f78deeb3254913bd05bf1f8c9a834168e9e2b89b/build.rs#L152-L160

**Reply by asmmo** [reply]
Hence `wasm32-unknown-unknown` has no place here. we must use `wasi` or `emscripten`, musn't we?

**Reply by jedisct1** [reply]
Libsodium requires either the `emscripten` or `wasi` environments, as it's not even possible to generate random numbers in standalone WebAssembly.
